nixos/modules/security/dhparams.nix

```diff
@@ -34,10 +40,19 @@ in
         type = types.str;
         default = "/var/lib/dhparams";
       };
+
+      enable = mkOption {
+        description =
+          ''
+            Whether to generate new DH params and clean up old DH params.
+          '';
+        default = false;
+        type = types.bool;
+      };
     };
   };
 
-  config.systemd.services = {
+  config.systemd.services = if !cfg.enable then {} else {
```

Comment:
Nit: The idiom is `config = mkIf cfg.enable { ... }`; or you can use `optionalAttrs`, which is a short-hand for the current code.